### PR TITLE
Setup react router & create action and reducer to change current product ID

### PR DESCRIPTION
**Pull request by shawnsull3**
- Setup the basics for the react router: honestly shouldn't change much since everything we have is a single page app, but it allows the client to utilize various browser functions like the back button.

- Eventually we can use the react router to display the product id in the url as well. ie: http://localhost:3000/5

- Setup an action, reducer, and type that can be used to change the current product id. I initialized it to be 1. @thomHayner you'll eventually be using this action/reducer for the related products section. They might need to be more built out, but I got them somewhat started (potentially functional)

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 import Reviews from './Components/Reviews/Reviews';
 
-function App() {
+function App(props) {
+  // console.log('props: ', props.location.pathname);
   return (
     <div className="App">
-      <header className="App-header">App</header>
-      <div className="components-area">
+      <div className="container">
```
Comment (shawnsull3) [inline src/App.jsx:13]: Changed to container so bootstrap will actually recognize our rows.

File: src/App.jsx [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 import './App.css';
+import { withRouter } from 'react-router-dom';
 import ProductOverview from './Components/ProductOverview/ProductOverview';
 import QuestionsAndAnswers from './Components/QuestionsAndAnswers/QuestionsAndAnswers';
 import RelatedProducts from './Components/RelatedProducts/RelatedProducts';
 import Reviews from './Components/Reviews/Reviews';
 
-function App() {
+function App(props) {
+  // console.log('props: ', props.location.pathname);
```
Comment (shawnsull3) [inline src/App.jsx:10]: @thomHayner commented out this line, but might be using prop.location.pathname to get a product id from the url in the future. keyword: MIGHT.

**Review by thomHayner — APPROVED** [review]
@shawnsull3 @jwhooperjr @morahan  I'm still quite confused about redux toolkit template.  I think that we're not supposed to use switch statements for reducers like in normal redux, but I'm not sure I don't know regular redux enough to definitively say.  If this is working and passing tests I can approve it so that we can move forward

**Comment by thomHayner** [comment]
@shawnsull3 @jwhooperjr @morahan I'm still quite confused about redux toolkit template. I think that we're not supposed to use switch statements for reducers like in normal redux, but I'm not sure I don't know regular redux enough to definitively say. If this is working and passing tests I can approve it so that we can move forward

**Pull request merged**